### PR TITLE
Engaging Crowds: Subject Picker: add Already Seen status

### LINKED ISSUE #1955: Engaging Crowds: personalise the subject selection tool
**Package**
app-project

**Is your feature request related to a problem? Please describe.**
When I'm logged-in, the subject selection tool should show me which subjects I've already classified, in the Status column of the table.

**Describe the solution you'd like**
The subjects table should make authenticated API requests, client-side, to check user-seen status for each subject listed. Retired subjects can be skipped. Subjects have three possible states (I think):

1. Completely retired.
2. Classified by you but not retired yet.
3. Not retired and not yet classified by you.

### PR DESCRIPTION
**Pull request by eatyourgreens**
Convert `checkRetirementStatus` to use the `subjects/selection` endpoint (with the current user's auth token), which gives us the already seen and retired status of each subject. Subjects have three possible statuses now: 'Unclassified', 'Already seen' and 'Retired'.

I've also localised the subject statuses so that they can be translated.

You can try it out on the Index or Minutes workflows for Scarlets & Blues, although those subjects are all unclassified at the moment.
http://localhost.zooniverse.org:3000/projects/bogden/scarlets-and-blues?env=production

Package:
app-project

Closes #1955.

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

**Review by camallen — COMMENTED** [review]
this looks like a great way to use the `subjects/selection?ids=1,2,3,4,5,6` end point to get the authenticated & non-authenticate user seen / retired subject states etc 🎩 

code LGTM as it's a change to the end point and adding the extra states to the data structures. But might be worth a frontend team member review to be sure

**Comment by eatyourgreens** [comment]
Scarlets & Blues only has a dozen or so subjects in each of its test sets. I imagine a real set would have at least 100 subjects, so I can see this code making ~10 subject selection requests when you first open the subjects table. Then more requests if you use the filtered search to request more subjects. We could cache those responses, so that searches don't make repeat requests for the same subject.

With Scarlets & Blues, I'm seeing two selection requests when the table first opens, and the responses seem pretty fast.

File: packages/app-project/src/shared/components/SubjectPicker/locales/en.json [hunk 1]
```diff
@@ -1,6 +1,9 @@
 {
   "SubjectPicker": {
     "heading": "Choose a subject to get started",
-    "byline": "Sort the list by clicking column names. You will see subjects sequentially, starting with the one you choose."
+    "byline": "Sort the list by clicking column names. You will see subjects sequentially, starting with the one you choose.",
+    "alreadySeen": "Already seen",
+    "retired": "Retired",
+    "unclassified": "Unclassified"
```
Comment (eatyourgreens) [inline packages/app-project/src/shared/components/SubjectPicker/locales/en.json:7]: @snblickhan should this say 'Unseen'?

Comment (eatyourgreens) [reply packages/app-project/src/shared/components/SubjectPicker/locales/en.json:7]: Or 'Available', after conversation on Slack.

Comment (eatyourgreens) [reply packages/app-project/src/shared/components/SubjectPicker/locales/en.json]: I've made that wording change, so this is ready now.

Comment (snblickhan) [reply packages/app-project/src/shared/components/SubjectPicker/locales/en.json]: Thanks! Yes, can confirm we'd settled on "Available" "Already seen" and "Retired" for the status options.

**Comment by shaunanoordin** [comment]
Code read looks good, heading into functionality tests now 👌

**Review by shaunanoordin — APPROVED** [review]
## PR Review

Package: `lib-classifier`
Context: the Engaging Crowds project intends to allow users to select Workflow -> Subject Set -> individual Subject

This PR updates the Subject selection component, so the list of Subjects also displays whether an individual Subject is available/retired/already seen, to save the volunteers' time.

_[Scarlet & Blues,](http://localhost:3000/projects/bogden/scarlets-and-blues/classify?env=production) WF 17077, SSet 96236, with randomly simulated retired/already seen status._

Screen Shot 2021-07-09 at 15 50 15

Code read looks good, and functionality checks out. LGTM! 👍 

### Testing

Tested `app-project` on localhost with macOS 10.12.6 + Chrome 91.

- URL: http://localhost:3000/projects/bogden/scarlets-and-blues/classify?env=production
- Select Workflow 17077 and Subject Set 96236
- The Subject selection menu shows as expected, with the correct "status"es of each Subject reported.

Testing took a bit longer than normal as I tried to debug why the Subject stasuses weren't correctly appearing for the other Subject Sets. 🤔 I'm still not sure why, but 1. I suspect it's related to the project-specific subject set shenanigans discovered yesterday and isn't related to this PR, and 2. I am very tired of trying to debug network calls that go through Next.js.

### Status

LGTM! 👍

**Comment by eatyourgreens** [comment]
>Testing took a bit longer than normal as I tried to debug why the Subject stasuses weren't correctly appearing for the other Subject Sets.

That's worrying. I'm also seeing the statuses as an empty column on workflow 92752.
http://local.zooniverse.org:3000/projects/bogden/scarlets-and-blues/classify/workflow/17077/subject-set/92752?env=production&demo=true

**Comment by eatyourgreens** [comment]
I'm going to merge this but keep an eye on why the statuses might not be working.

**Comment by eatyourgreens** [comment]
I think those subject sets have been removed from the workflow, in the project builder, but are still showing up in the response from Cellect. https://cellect.zooniverse.org/workflows/17077/status

That's a definite weak point in the code.

**Pull request merged**